docker-compose.stack.yml

```diff
@@ -0,0 +1,138 @@
+version: "3.7"
+
+x-custom:
+  shared_image: &shared_image
+    image: smirart/postgres-xl:kubernetes
+
+services:
+  pgpool:
```

Comment:
So, Pgpool… Pgpool is excellent. But why do we need it? It's already possible to address any coordinator, and if there aren't enough spare connections, it's possible to simply add a coordinator (since all the topologies in this project, including the `init-eg-*` scripts, and the YAML files, are examples only). So this adds another piece to the stack, which I don't automatically understand is necessary. If it's for load-balancing, there are other ways of achieving that beyond Pgpool—e.g. what if someone wants to use HAProxy instead? If it's for high-availability—well, this doesn't give it, because the Pgpool instance could also fail, and there's only one of them. And if we're relying on Kubernetes to detect the health and relaunch it—isn't that what will happen for a coordinator anyway?

Reply:
As I could understand when we use two or more coordinators we haven't one enter point (connection point). `pgpool` may be that enter point. Therefore I added him. But how I've agreed above, `pgpool` is optional.

Reply:
But that still doesn't solve the issue, does it? Because unless you have multiple instances of Pgpool, there's still no redundancy there—unless you're relying on unhealthy containers being killed and relaunched. Which I think is reasonable—except that that's arguably what happens for coordinators anyway. And even if someone wanted a single entrypoint for a connection, they might be using another method entirely—such as HAProxy, or DNS failover, or a VIP. There's no reason it has to be Pgpool.